pylint/checkers/misc.py

```diff
@@ -133,16 +150,43 @@ def process_tokens(self, tokens: list[tokenize.TokenInfo]) -> None:
         if not self.linter.config.notes:
             return
         for token_info in tokens:
-            if token_info.type != tokenize.COMMENT:
-                continue
-            comment_text = token_info.string[1:].lstrip()  # trim '#' and white-spaces
-            if self._fixme_pattern.search("#" + comment_text.lower()):
-                self.add_message(
-                    "fixme",
-                    col_offset=token_info.start[1] + 1,
-                    args=comment_text,
-                    line=token_info.start[0],
-                )
+            if token_info.type == tokenize.COMMENT:
+                comment_text = token_info.string[
+                    1:
+                ].lstrip()  # trim '#' and white-spaces
+                if self._comment_fixme_pattern.search("#" + comment_text.lower()):
+                    self.add_message(
+                        "fixme",
+                        col_offset=token_info.start[1] + 1,
+                        args=comment_text,
+                        line=token_info.start[0],
+                    )
+            elif (
+                self.linter.config.check_fixme_in_docstring
+                and self._is_docstring_comment(token_info)
+            ):
+                docstring_lines = token_info.string.split("\n")
+                for line_no, line in enumerate(docstring_lines):
+                    # trim '"""' at beginning or end and whitespace
+                    if line.startswith(('"""', "'''")):
+                        line = line[3:]
+                    line = line.lstrip()
+                    if line.endswith(('"""', "'''")):
+                        line = line[:-3]
+                    if self._docstring_fixme_pattern.search(
+                        '"""' + line.lower()
+                    ) or self._docstring_fixme_pattern.search("'''" + line.lower()):
```

Comment:
Now that you have fixed the pattern, is this really necessary? I would prefer a pattern where we don't have to do a lot of complicated changes to the lines itself.

Reply:
True, it was getting pretty messy. I've decided to rework the logic to use regex capture groups for both docstrings **and** comments (since in that case we were also sort of changing the line then stitching it together to examine with regex).